File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.js
/**
 * @param {string[]} words
 * @return {number}
 */
var maximumNumberOfStringPairs = function(words) {
    const strings= new Set()
    let count=0
    for (let word of words){
        if(strings.has(word)){
            count++
        }else{
            strings.add(word.split("").reverse().join(""))
        }
    }
    return count
};

File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.py
class Solution:
    def maximumNumberOfStringPairs(self, words: List[str]) -> int:
        count=0
        n=len(words)
        for i in range(n):
            for j in range(i+1,n):
                if words[i]==words[j][::-1]:
                    count+=1
        return count